--- gdl-0.9pre6/src/antlr/CharScanner.hpp.gcc43	2007-10-29 17:10:45.000000000 -0600
+++ gdl-0.9pre6/src/antlr/CharScanner.hpp	2008-02-13 10:55:14.000000000 -0700
@@ -10,6 +10,8 @@
 
 #include <antlr/config.hpp>
 
+#include <cstdlib>
+#include <cstring>
 #include <map>
 
 #ifdef HAS_NOT_CCTYPE_H
--- gdl-0.9pre6/src/antlr/Parser.cpp.gcc	2007-10-29 17:10:45.000000000 -0600
+++ gdl-0.9pre6/src/antlr/Parser.cpp	2008-02-13 11:18:20.000000000 -0700
@@ -11,6 +11,7 @@
 #include "antlr/TokenBuffer.hpp"
 #include "antlr/MismatchedTokenException.hpp"
 //#include "antlr/ASTFactory.hpp"
+#include <cstdlib>
 #include <iostream>
 
 #ifdef ANTLR_CXX_SUPPORTS_NAMESPACE
--- gdl-0.9pre6/src/antlr/TreeParser.cpp.gcc43	2007-10-29 17:10:45.000000000 -0600
+++ gdl-0.9pre6/src/antlr/TreeParser.cpp	2008-02-13 11:20:28.000000000 -0700
@@ -8,6 +8,7 @@
 #include "antlr/TreeParser.hpp"
 #include "antlr/ASTNULLType.hpp"
 #include "antlr/MismatchedTokenException.hpp"
+#include <cstdlib>
 #include <iostream>
 #include <stdio.h>
 
--- gdl-0.9pre6/src/basic_pro.cpp.gcc43	2007-10-29 17:10:47.000000000 -0600
+++ gdl-0.9pre6/src/basic_pro.cpp	2008-02-13 11:30:20.000000000 -0700
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <fstream>
+#include <iterator>
 #include <memory>
 #include <set>
 
