--- gdl-0.9rc1/src/antlr/TreeParser.cpp.gcc43	2008-04-03 16:09:26.000000000 -0600
+++ gdl-0.9rc1/src/antlr/TreeParser.cpp	2008-09-05 22:06:20.000000000 -0600
@@ -8,6 +8,7 @@
 #include "antlr/TreeParser.hpp"
 #include "antlr/ASTNULLType.hpp"
 #include "antlr/MismatchedTokenException.hpp"
+#include <cstdlib>
 #include <iostream>
 #include <stdio.h>
 
--- gdl-0.9rc1/src/antlr/CharScanner.hpp.gcc43	2008-09-05 22:06:20.000000000 -0600
+++ gdl-0.9rc1/src/antlr/CharScanner.hpp	2008-09-05 22:06:20.000000000 -0600
@@ -14,6 +14,8 @@
 #include <cstring>
 #include <cstdlib>
 
+#include <cstdlib>
+#include <cstring>
 #include <map>
 
 #ifdef HAS_NOT_CCTYPE_H
--- gdl-0.9rc1/src/antlr/Parser.cpp.gcc43	2008-04-03 16:09:26.000000000 -0600
+++ gdl-0.9rc1/src/antlr/Parser.cpp	2008-09-05 22:06:20.000000000 -0600
@@ -11,6 +11,7 @@
 #include "antlr/TokenBuffer.hpp"
 #include "antlr/MismatchedTokenException.hpp"
 //#include "antlr/ASTFactory.hpp"
+#include <cstdlib>
 #include <iostream>
 
 #ifdef ANTLR_CXX_SUPPORTS_NAMESPACE
--- gdl-0.9rc1/src/basic_pro.cpp.gcc43	2008-09-05 22:06:20.000000000 -0600
+++ gdl-0.9rc1/src/basic_pro.cpp	2008-09-05 22:06:20.000000000 -0600
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <fstream>
+#include <iterator>
 #include <memory>
 #include <set>
 #include <iterator>
