diff -up gdl-0.9.2/testsuite/try.build gdl-0.9.2/testsuite/try
--- gdl-0.9.2/testsuite/try.build	2011-05-02 05:00:47.000000000 -0600
+++ gdl-0.9.2/testsuite/try	2012-05-10 16:16:14.218180027 -0600
@@ -13,8 +13,8 @@
 # - the procedure should not need any arguments/keywords
 
 # a quick & dirty way of making the "checks" script work
-test "x$2" = "x./checks" && exit
+test "x$3" = "x./checks" && exit
 
 echo "exit,status=1" | \
-  IDL_PATH="" IDL_STARTUP="" GDL_STARTUP="" GDL_PATH=+$1/src/pro:$1/testsuite \
-  $1/src/gdl -quiet -e `echo $2 | cut -d/ -f2 | cut -d. -f1` &> /dev/null
+  IDL_PATH="" IDL_STARTUP="" GDL_STARTUP="" GDL_PATH=+$2/src/pro:$1/testsuite \
+  $2/src/gdl -quiet -e `echo $2 | cut -d/ -f2 | cut -d. -f1` &> /dev/null
