diff -up gdl-0.9.8/src/dinterpreter.cpp.sign gdl-0.9.8/src/dinterpreter.cpp
--- gdl-0.9.8/src/dinterpreter.cpp.sign	2018-04-14 20:29:23.612484457 -0600
+++ gdl-0.9.8/src/dinterpreter.cpp	2018-04-14 20:29:23.628484629 -0600
@@ -666,7 +666,7 @@ DInterpreter::CommandCode DInterpreter::
 DInterpreter::CommandCode DInterpreter::CmdCompile( const string& command)
 {
   string cmdstr = command;
-  int sppos = cmdstr.find(" ",0);
+  size_t sppos = cmdstr.find(" ",0);
   if (sppos == string::npos) 
     {
       cout << "Interactive COMPILE not implemented yet." << endl;
@@ -676,7 +676,7 @@ DInterpreter::CommandCode DInterpreter::
   bool retAll = false; // Remember if Retall is needed
 
   // Parse each file name
-  int pos = sppos + 1;
+  size_t pos = sppos + 1;
   while (pos < command.length()) 
     {
       sppos = command.find(" ",pos);
@@ -737,7 +737,7 @@ DInterpreter::CommandCode DInterpreter::
   bool retAll = false; // Remember if Retall is needed
 
   // Parse each file name
-  int pos = sppos + 1;
+  size_t pos = sppos + 1;
   while (pos < command.length()) 
     {
       sppos = command.find(" ",pos);
@@ -799,7 +799,7 @@ DInterpreter::CommandCode DInterpreter::
 {
   string cmdstr = command;
   string args;
-  int sppos = cmdstr.find(" ",0);
+  size_t sppos = cmdstr.find(" ",0);
   if (sppos != string::npos) {
     args = cmdstr.substr(sppos+1);
     cmdstr = cmdstr.substr(0, sppos);
diff -up gdl-0.9.8/src/GDLLexer.cpp.sign gdl-0.9.8/src/GDLLexer.cpp
--- gdl-0.9.8/src/GDLLexer.cpp.sign	2018-03-26 03:57:27.000000000 -0600
+++ gdl-0.9.8/src/GDLLexer.cpp	2018-04-14 20:30:05.656936102 -0600
@@ -434,7 +434,7 @@ void GDLLexer::mINCLUDE(bool _createToke
 				std::string name = f->getText();
 		
 		// find comments on the same line
-		long pos = name.find_first_of(';', 0);   
+		size_t pos = name.find_first_of(';', 0);   
 		if( pos != std::string::npos) // remove them  
 		name = name.substr(0, pos);
 		
diff -up gdl-0.9.8/src/GDLParser.cpp.sign gdl-0.9.8/src/GDLParser.cpp
--- gdl-0.9.8/src/GDLParser.cpp.sign	2018-03-26 03:57:27.000000000 -0600
+++ gdl-0.9.8/src/GDLParser.cpp	2018-04-14 20:29:23.630484650 -0600
@@ -6773,7 +6773,7 @@ void GDLParser::arrayindex_list() {
 	antlr::ASTPair currentAST;
 	RefDNode arrayindex_list_AST = RefDNode(antlr::nullAST);
 			
-	int rank = 1;
+	SizeT rank = 1;
 	
 	
 	if ((LA(1) == LSQUARE)) {
