--- gdl-0.9rc2-20090312/configure.antlr	2009-03-12 08:41:10.000000000 -0600
+++ gdl-0.9rc2-20090312/configure	2009-03-12 08:42:04.000000000 -0600
@@ -22102,7 +22102,7 @@
 
 
 
-ac_config_files="$ac_config_files Makefile src/Makefile src/antlr/Makefile"
+ac_config_files="$ac_config_files Makefile src/Makefile"
 
 
 cat >confcache <<\_ACEOF
@@ -22707,7 +22707,6 @@
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
     "src/Makefile") CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
-    "src/antlr/Makefile") CONFIG_FILES="$CONFIG_FILES src/antlr/Makefile" ;;
 
   *) { { echo "$as_me:$LINENO: error: invalid argument: $ac_config_target" >&5
 echo "$as_me: error: invalid argument: $ac_config_target" >&2;}
--- gdl-0.9rc2-20090312/src/Makefile.in.antlr	2009-02-23 11:24:00.000000000 -0700
+++ gdl-0.9rc2-20090312/src/Makefile.in	2009-02-26 09:00:54.000000000 -0700
@@ -55,7 +55,7 @@
 libLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(lib_LTLIBRARIES)
 @BUILDASLIBRARY_TRUE@libgdl_la_DEPENDENCIES =  \
-@BUILDASLIBRARY_TRUE@	$(top_builddir)/src/antlr/libantlr.la
+@BUILDASLIBRARY_TRUE@	-lantlr
 am__libgdl_la_SOURCES_DIST = accessdesc.hpp arrayindex.hpp \
 	assocdata.cpp assocdata.hpp basegdl.hpp basic_fun_cl.cpp \
 	basic_fun_cl.hpp basic_fun.cpp basic_fun.hpp basic_fun_jmg.cpp \
@@ -233,7 +233,7 @@
 @BUILDASLIBRARY_FALSE@	gdl-gdl.$(OBJEXT)
 gdl_OBJECTS = $(am_gdl_OBJECTS)
 @BUILDASLIBRARY_FALSE@gdl_DEPENDENCIES =  \
-@BUILDASLIBRARY_FALSE@	$(top_builddir)/src/antlr/libantlr.a
+@BUILDASLIBRARY_FALSE@	-lantlr
 gdl_LINK = $(LIBTOOL) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(CXXLD) $(gdl_CXXFLAGS) $(CXXFLAGS) \
 	$(gdl_LDFLAGS) $(LDFLAGS) -o $@
@@ -391,7 +391,6 @@
 
 # set the include path found by configure
 INCLUDES = $(EXT_INCLUDES) $(all_includes)
-SUBDIRS = antlr
 noinst_HEADERS = gdlwidget.hpp prognode.hpp prognodeexpr.hpp
 cpp_hpp_files = accessdesc.hpp arrayindex.hpp assocdata.cpp assocdata.hpp \
 		basegdl.hpp basic_fun_cl.cpp basic_fun_cl.hpp basic_fun.cpp basic_fun.hpp \
@@ -438,7 +437,7 @@
 @BUILDASLIBRARY_FALSE@gdl_CXXFLAGS = $(AM_CXXFLAGS)
 # the library search path.
 @BUILDASLIBRARY_FALSE@gdl_LDFLAGS = $(all_libraries)
-@BUILDASLIBRARY_FALSE@gdl_LDADD = $(top_builddir)/src/antlr/libantlr.a -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
+@BUILDASLIBRARY_FALSE@gdl_LDADD = -lantlr -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
 
 # build as library (for python module)
 @BUILDASLIBRARY_TRUE@lib_LTLIBRARIES = libgdl.la
@@ -448,7 +447,7 @@
 # netCDF and HDF define some same symbols
 # hence -z muldefs has to be passed to the linker
 @BUILDASLIBRARY_TRUE@@DARWIN_TRUE@libgdl_la_LDFLAGS = $(all_libraries) -Wl,-m
-@BUILDASLIBRARY_TRUE@libgdl_la_LIBADD = $(top_builddir)/src/antlr/libantlr.la -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
+@BUILDASLIBRARY_TRUE@libgdl_la_LIBADD = -lantlr -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
 all: all-recursive
 
 .SUFFIXES:
