diff -up gdl-0.9.3/src/Makefile.am.shared gdl-0.9.3/src/Makefile.am
--- gdl-0.9.3/src/Makefile.am.shared	2012-12-27 09:22:44.000000000 -0700
+++ gdl-0.9.3/src/Makefile.am	2013-02-25 16:59:51.886326564 -0700
@@ -78,7 +78,7 @@ libgdl_la_SOURCES = $(cpp_hpp_files) pyt
 if DARWIN
 libgdl_la_LDFLAGS = $(all_libraries) -Wl,-m
 else
-libgdl_la_LDFLAGS = $(all_libraries) -Wl,-z,muldefs
+libgdl_la_LDFLAGS = -rpath $(libdir) $(all_libraries) -Wl,-z,muldefs
 endif 
 
 libgdl_la_LIBADD = $(top_builddir)/src/antlr/libantlr.la -L/usr/X11R6/lib64 -L/usr/X11R6/lib -lX11
