--- gdl-0.9-cvs/CMakeLists.txt.hdf	2011-03-18 13:26:25.000000000 -0600
+++ gdl-0.9-cvs/CMakeLists.txt	2011-03-18 14:53:43.809165287 -0600
@@ -326,7 +326,7 @@
 	if(HDF_FOUND)
 		if(NETCDF)
 			set(CMAKE_REQUIRED_LIBRARIES ${HDF_EXTRA_LIBRARIES})
-			check_library_exists(${HDF_LIBRARIES} sd_nccreate "" SD_NCCREATE)
+			check_library_exists("${HDF_LIBRARIES}" sd_nccreate "" SD_NCCREATE)
 			if(NOT SD_NCCREATE)
 				message(FATAL_ERROR "HDF4 needs to be configured with the --disable-netcdf option "
 				"in order to be used with the original netCDF library.")
