diff -up gdl-0.9.4/testsuite/test_matrix_multiply.pro.gsl gdl-0.9.4/testsuite/test_matrix_multiply.pro
--- gdl-0.9.4/testsuite/test_matrix_multiply.pro.gsl	2013-04-11 10:00:30.000000000 -0600
+++ gdl-0.9.4/testsuite/test_matrix_multiply.pro	2013-10-02 11:21:57.504511516 -0600
@@ -39,7 +39,7 @@ identity=DIAG_MATRIX(REPLICATE(1., nbp))
 matrice=DIAG_MATRIX(REPLICATE(value, nbp-1), 1)
 matrice[0,nbp-1]=value
 ;
-inv_matrice=INVERT(matrice)
+inv_matrice=INVERT(matrice,status)
 ;
 ; We use a "rotation matrix"
 ;
@@ -70,8 +70,15 @@ if (errors GT 0) then begin
     print, 'effective type : ', effective_type
     nb_errors=nb_errors+errors
 endif
-;
+
 if KEYWORD_SET(debug) then begin
+    print, 'Invert status = ', status
+    print, 'Where inv_matrice ne 0 : ', where(inv_matrice ne 0)
+    print, 'Where trans_matrice ne 0 : ', where(trans_matrice ne 0)
+    print, 'Where test1 ne 0 : ', where(abs(test1) gt tolerance)
+    print, 'Where test2 ne 0 : ', where(abs(test2) gt tolerance)
+    print, 'Where test3 ne 0 : ', where(abs(test3) gt tolerance)
+    print, 'Where test4 ne 0 : ', where(abs(test4) gt tolerance)
     print, MIN(test1), MAX(test1)
     print, MIN(test2), MAX(test2)
     print, MIN(test3), MAX(test3)
@@ -110,7 +117,7 @@ liste_taille=[2,3,4,10,11,13,25,50,51,10
 for itailles=0, N_ELEMENTS(liste_taille)-1 do begin
     for itypes=0, N_ELEMENTS(liste_type)-1 do begin
         TEST_MATRIX_UNITARY, type=liste_type[itypes], $
-          nbp=liste_taille[itailles], nb_errors=nb_errors, verbose=verbose
+          nbp=liste_taille[itailles], nb_errors=nb_errors, verbose=verbose, debug=1
     endfor
 endfor
 ;
