--- gdl-0.9-cvs/CMakeModules/FindNetCDF.cmake.netcdf	2010-07-16 07:26:09.000000000 -0600
+++ gdl-0.9-cvs/CMakeModules/FindNetCDF.cmake	2011-03-18 14:27:47.993962330 -0600
@@ -6,7 +6,7 @@
 	pkg_check_modules(NETCDF netcdf)
 	if(NETCDF_FOUND)
 		set(NETCDF_INCLUDE_DIR ${NETCDF_INCLUDE_DIRS})
-		find_package_handle_standard_args(NETCDF DEFAULT_MSG NETCDF_LIBRARIES NETCDF_INCLUDE_DIR)
+		find_package_handle_standard_args(NETCDF DEFAULT_MSG NETCDF_LIBRARIES)
 	endif(NETCDF_FOUND)
 else(NETCDFPKG) # no netcdf.pc file
 	find_library(NETCDF_LIBRARIES NAMES netcdf)
