diff -up gdl-0.9.8/src/saverestore.cpp.return gdl-0.9.8/src/saverestore.cpp
--- gdl-0.9.8/src/saverestore.cpp.return	2018-03-26 03:57:27.000000000 -0600
+++ gdl-0.9.8/src/saverestore.cpp	2018-04-11 16:33:39.129920693 -0600
@@ -164,6 +164,7 @@ namespace lib {
   
   uint32_t writeEnd(XDR *xdrs) {
     uint32_t cur=writeNewRecordHeader(xdrs, 6);
+    return cur;
   }
 
   int getVersion(XDR* xdrs) {
