--- gdl-0.9pre6/src/plot3d_nr.cpp.plplot	2007-10-29 17:10:47.000000000 -0600
+++ gdl-0.9pre6/src/plot3d_nr.cpp	2008-02-27 09:39:56.000000000 -0700
@@ -217,7 +217,7 @@
     zmin = (zmax_in > zmin_in) ? zmin_in: zmax_in;
     zmax = (zmax_in > zmin_in) ? zmax_in: zmin_in;
 
-    pldtik(zmin, zmax, &tick, &nsub);
+    pldtik(zmin, zmax, &tick, &nsub, FALSE);
     tp = tick * floor(zmin / tick) + tick;
     pl3upv = 0;
 
