--- gdl-0.9rc4/configure.in.wx-config	2010-02-02 11:13:51.000000000 -0700
+++ gdl-0.9rc4/configure.in	2010-02-03 15:43:32.849333374 -0700
@@ -322,23 +322,8 @@
     dnl SA: this is needed to compile on Leopard
     dnl (see e.g. the "OpenGL bug section" in http://wiki.finkproject.org/index.php/Fink:Packaging:Preparing_for_10.5#OpenGL_Bug)
     dnl LDFLAGS="$LDFLAGS -Wl,-dylib_file,/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib:/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib"
+    AC_DEFINE([HAVE_LIBWXWIDGETS], [1], [wxWidgets library])
   fi
-
-  tmp="`wx-config --basename`-`wx-config --release`"
-  AC_CHECK_LIB($tmp, wxFatalSignalHandler, 
-    [AC_DEFINE([HAVE_LIBWXWIDGETS], [1], [wxWidgets library])], [
-      dnl SA: handling the multi-port wxWidgets installation case	 
-      tmp="`wx-config --basename --toolkit=base`-`wx-config --release`"	 
-      AC_CHECK_LIB($tmp, wxInteger_compare, [AC_DEFINE([HAVE_LIBWXWIDGETS], [1])], [
-        echo ""
-        echo "Error! wxWidgets version 2.8 or later is required but was not found"
-        echo "       Use --with-wxWidgets=DIR to specify the wxWidgets directory tree"
-        echo "       Use --with-wxWidgets=no  to not use it"
-        echo "       Check the README or use configure --help for other libraries needed"
-        echo "       (--with-xxxdir = obligatory, --with-xxx = optional (--with-xxx=no to disable))"
-        exit -1
-      ])
-  ])
 fi
 
 dnl == ImageMagick =========================================
