diff -up gdl-1.0.2/src/CMakeLists.txt.antlr gdl-1.0.2/src/CMakeLists.txt
--- gdl-1.0.2/src/CMakeLists.txt.antlr	2023-01-06 10:39:47.000000000 -0700
+++ gdl-1.0.2/src/CMakeLists.txt	2023-01-16 08:51:18.033486027 -0700
@@ -197,9 +197,7 @@ endif(HAVE_LIBWXWIDGETS AND WIN32)
 #local plplot drivers
 add_subdirectory(plplotdriver)
 
-add_subdirectory(antlr)
-
-include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src/antlr ${CMAKE_SOURCE_DIR}/src/plplotdriver ${CMAKE_BINARY_DIR})
+include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src/plplotdriver ${CMAKE_BINARY_DIR})
 link_directories(${LINK_DIRECTORIES})
 
 if(PYTHON_MODULE) #GDL.so
@@ -217,7 +215,6 @@ if(USE_OPENMP)
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 endif(USE_OPENMP)
 
-add_dependencies(gdl antlr) # be sure that antlr is built before gdl
 target_link_libraries(gdl antlr) # link antlr against gdl
 if (MINGW)
 target_link_libraries(gdl ws2_32)
