diff -up gdl-0.9-cvs/CMakeLists.txt.config gdl-0.9-cvs/CMakeLists.txt
--- gdl-0.9-cvs/CMakeLists.txt.config	2011-03-18 16:34:56.468675812 -0600
+++ gdl-0.9-cvs/CMakeLists.txt	2011-03-18 16:38:48.122620250 -0600
@@ -543,7 +543,7 @@ install(FILES ${CMAKE_SOURCE_DIR}/AUTHOR
 install(FILES ${CMAKE_SOURCE_DIR}/doc/gdl.1 DESTINATION ${CMAKE_INSTALL_PREFIX}/share/man/man1)
 
 # substitute variables in configure.h.cmake and move it to configure.h
-configure_file(${CMAKE_SOURCE_DIR}/config.h.cmake ${CMAKE_SOURCE_DIR}/config.h)
+configure_file(${CMAKE_SOURCE_DIR}/config.h.cmake ${CMAKE_BINARY_DIR}/config.h)
 
 # create a link src/gdl -> ${BUILD}/.../gdl
 if(NOT PYTHON_MODULE)
diff -up gdl-0.9-cvs/src/CMakeLists.txt.config gdl-0.9-cvs/src/CMakeLists.txt
--- gdl-0.9-cvs/src/CMakeLists.txt.config	2011-03-18 16:34:56.491673128 -0600
+++ gdl-0.9-cvs/src/CMakeLists.txt	2011-03-18 16:38:39.272652389 -0600
@@ -203,7 +203,7 @@ typetraits.hpp
 widget.cpp
 )
 
-include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src)
+include_directories(${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_BINARY_DIR})
 link_directories(${LINK_DIRECTORIES})
 
 if(PYTHON_MODULE) #libgdl
