diff -up gdl-0.9.8/src/medianfilter.cpp.vector gdl-0.9.8/src/medianfilter.cpp
--- gdl-0.9.8/src/medianfilter.cpp.vector	2018-03-26 03:57:27.000000000 -0600
+++ gdl-0.9.8/src/medianfilter.cpp	2018-05-20 15:19:38.072948145 -0600
@@ -673,8 +673,8 @@ static inline void histogram_add( const
 #elif defined(__ALTIVEC__)
 static inline void histogram_add( const uint16_t x[16], uint16_t y[16] )
 {
-    *(vector unsigned short*) &y[0] = vec_add( *(vector unsigned short*) &y[0], *(vector unsigned short*) &x[0] );
-    *(vector unsigned short*) &y[8] = vec_add( *(vector unsigned short*) &y[8], *(vector unsigned short*) &x[8] );
+    *(__vector unsigned short*) &y[0] = vec_add( *(__vector unsigned short*) &y[0], *(__vector unsigned short*) &x[0] );
+    *(__vector unsigned short*) &y[8] = vec_add( *(__vector unsigned short*) &y[8], *(__vector unsigned short*) &x[8] );
 }
 #else
 static inline void histogram_add( const uint16_t x[16], uint16_t y[16] )
@@ -707,8 +707,8 @@ static inline void histogram_sub( const
 #elif defined(__ALTIVEC__)
 static inline void histogram_sub( const uint16_t x[16], uint16_t y[16] )
 {
-    *(vector unsigned short*) &y[0] = vec_sub( *(vector unsigned short*) &y[0], *(vector unsigned short*) &x[0] );
-    *(vector unsigned short*) &y[8] = vec_sub( *(vector unsigned short*) &y[8], *(vector unsigned short*) &x[8] );
+    *(__vector unsigned short*) &y[0] = vec_sub( *(__vector unsigned short*) &y[0], *(__vector unsigned short*) &x[0] );
+    *(__vector unsigned short*) &y[8] = vec_sub( *(__vector unsigned short*) &y[8], *(__vector unsigned short*) &x[8] );
 }
 #else
 static inline void histogram_sub( const uint16_t x[16], uint16_t y[16] )
