diff -up gdl-0.9.3/testsuite/test_ce.pro.test_ce gdl-0.9.3/testsuite/test_ce.pro
--- gdl-0.9.3/testsuite/test_ce.pro.test_ce	2012-12-27 09:22:44.000000000 -0700
+++ gdl-0.9.3/testsuite/test_ce.pro	2013-03-11 16:27:20.624137669 -0600
@@ -6,7 +6,7 @@ pro test_ce
     ; should be used together with test_ce.so compiled from test_ce.c
     ;-----------------------------------------------------------------------
 
-    image = 'libtest_ce.' + (!VERSION.OS_NAME eq 'Darwin' ? "dylib" : "so")
+    image = './libtest_ce.' + (!VERSION.OS_NAME eq 'Darwin' ? "dylib" : "so")
     if ~file_test(image) then image = '.libs/' + image
     if ~file_test(image) then begin
       message, 'library found not compiled!', /conti
