--- gdl-0.9rc1/configure.magick	2008-05-16 11:45:56.000000000 -0600
+++ gdl-0.9rc1/configure	2008-05-16 11:48:06.000000000 -0600
@@ -20655,16 +20655,16 @@
     INCLUDES="$INCLUDES -I$with_Magick/include"
   else
     LIBS="$LIBS -lMagick++"
-    INCLUDES="$INCLUDES -I/usr/include/magick"
+    INCLUDES="$INCLUDES -I/usr/include/ImageMagick"
   fi
 
-  { echo "$as_me:$LINENO: checking for GetMagickVersion in -lMagick" >&5
-echo $ECHO_N "checking for GetMagickVersion in -lMagick... $ECHO_C" >&6; }
+  { echo "$as_me:$LINENO: checking for GetMagickVersion in -lMagick++" >&5
+echo $ECHO_N "checking for GetMagickVersion in -lMagick++... $ECHO_C" >&6; }
 if test "${ac_cv_lib_Magick_GetMagickVersion+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lMagick  $LIBS"
+LIBS="-lMagick++  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
