diff -up gdl-0.9.3/CMakeLists.txt.python gdl-0.9.3/CMakeLists.txt
--- gdl-0.9.3/CMakeLists.txt.python	2013-03-21 14:21:33.649335754 -0600
+++ gdl-0.9.3/CMakeLists.txt	2013-03-21 14:33:55.738083209 -0600
@@ -562,6 +562,11 @@ if(PYTHON OR PYTHON_MODULE)
                   message("-- Found Python executable: ${PYTHON_EXECUTABLE}")
                 endif()
 	else()
+		if(PYTHONVERSION)
+			set(PythonLibs_FIND_VERSION ${PYTHONVERSION})
+		else()
+			set(PythonLibs_FIND_VERSION 2)
+		endif()
 		find_package(PythonLibs)
                 include(FindPythonInterp)
 	endif()
